Fix SQL errors in initialization of tracks that are already filtered out

cdamus opened the pull request:
The process summary tracks can have dynamically defined tables that they create dropped during the sequence of asynchronous initialization steps in which they create those tables and query them.

Implement an abstraction to simplify the management of these tables:
- track the created table and existence thereof
- easily drop the table on destruction of the track controller
- execute queries on the table if and only if it exists, with fall-back options for when it doesn't

Apply the DynamicTable mechanism to manage dynamic tables for the process summary tracks and also the counter tracks that previously neglected to drop their tables (thus fixing a latent bug).

This PR fixes issue android-graphics/sokatoa#2317.

----

To test:

1. Open a capture that has Perfetto data.
2. Filter out all tracks except for the Memory Usage group of tracks. In particular, it's important to filter out all process track groups.
3. See that the timeline is greatly reduced in the number of tracks (there should now be a half dozen or so only) and there are no errors reported.
4. Close and re-open the capture.
5. Verify that it looks as it did at step 3 and there are no errors reported.

cdamus committed:
Fix SQL errors in initialization of tracks that are already filtered out

The process summary tracks can have dynamically defined tables that they
create dropped during the sequence of asynchronous initialization steps
in which they create those tables and query them.

Implement an abstraction to simplify the management of these tables:
- track the created table and existence thereof
- easily drop the table on destruction of the track controller
- execute queries on the table if and only if it exists, with fall-back
  options for when it doesn't

Apply the DynamicTable mechanism to manage dynamic tables for the
process summary tracks and also the counter tracks that previously
neglected to drop their tables (thus fixing a latent bug).

Signed-off-by: Christian W. Damus <cdamus.ext@eclipsesource.com>

jfaltermeier commented on ui/src/controller/track_controller.ts:
```diff
@@ -277,6 +374,91 @@ export abstract class TrackController<
+   * @return {Data} a token track-data object signalling cancellation
+   */
+  protected cancelData(): Data {
+    return {[cancelToken]: cancelToken} as unknown as Data;
```
Just a question: the `as unknown as Data` looks a bit dangerous. I assume one reason for doing it this way is to avoid introducing something like `Data | CancellationToken` types throughout the code, which could make the code harder to use. Were there any other considerations?

cdamus replied:
Nope, that was it, avoiding a larger refactoring. I happen to know that this will be used only in initialization of a controller for a track that is being filtered out, so the broken `Data` object will not matter. I should add some further explanatory comments to make sure that this will not be used in any other context. Thanks for raising this!

cdamus replied:
Actually @jfaltermeier on reflection you are absolutely correct that we do not need this special cancellation token. I've cancelled it in commit 75ba5c5508ecf82d0d3a836db49de87053799773 😀  There is a perfectly reasonable empty data object that we can publish for both kinds of tracks involved in this PR.

cdamus committed:
Cancel the "Cancel Data" concept

We do not need a special cancellation token in place of otherwise valid
data to signal interruption of the track controller initialization. The
empty data object can be published as is because the track will be
filtered out, anyways.

Signed-off-by: Christian W. Damus <cdamus.ext@eclipsesource.com>
